src/Pact/Analyze/Eval/Term.hs

```diff
@@ -133,23 +136,35 @@ tagAccessCell lens' tid fieldName av = do
   mTag <- preview $
     aeModelTags.lens'.at tid._Just.located._2.objFields.at fieldName._Just._2
   case mTag of
-    -- NOTE: ATM we allow a "partial" model. we could also decide to
-    -- 'throwError' here; we simply don't tag.
     Nothing    -> pure ()
     Just tagAv -> addConstraint $ sansProv $ av .== tagAv
 
+tagAssert :: TagId -> S Bool -> Analyze ()
+tagAssert tid sb = do
+  mTag <- preview $ aeModelTags.mtAsserts.at tid._Just.located
+  case mTag of
+    Nothing  -> pure ()
+    Just sbv -> addConstraint $ sansProv $ sbv .== _sSbv sb
+
 -- | "Tag" an uninterpreted auth value with value from our Model that was
 -- allocated in Symbolic.
-tagAuth :: TagId -> Maybe Provenance -> S Bool -> Analyze ()
-tagAuth tid mKsProv sb = do
-  mTag <- preview $ aeModelTags.mtAuths.at tid._Just.located
+tagAuth :: TagId -> S KeySet -> S Bool -> Analyze ()
+tagAuth tid sKs sb = do
+  mTup <- preview $ aeModelTags.mtAuths.at tid._Just.located
+  case mTup of
+    Nothing  -> pure ()
+    Just (ksTag, sbv) -> do
+      addConstraint $ sansProv $ ksTag .== sKs
+      addConstraint $ sansProv $ sbv .== _sSbv sb
+      globalState.gasKsProvenances.at tid .= (sKs ^. sProv)
+
+tagSubpathStart :: S Bool -> TagId -> Analyze ()
+tagSubpathStart sb tid = do
```

Comment:
Instead of `sb`, should this be called something like `reachable`?

Reply:
There seems to be a difference between this case and, eg, `tagAssert`, where the `sb` in `tagAssert` means "assertion passes", whereas here it means "reachable". Is that right?

Reply:
👍